File: ExamenFinal/app/src/main/java/com/example/examenfinal/backend/BaseDatos.java
package com.example.examenfinal.backend;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

public class BaseDatos extends SQLiteOpenHelper {
    private static final String DATABASE_NAME = "users.db";
    private static final int DATABASE_VERSION = 1;

    private static final String TABLE_USERS = "users";
    private static final String COLUMN_ID = "id";
    private static final String COLUMN_NAME = "name";
    private static final String COLUMN_EMAIL = "email";
    private static final String COLUMN_GENDER = "gender";
    private static final String COLUMN_STATUS = "status";

    public BaseDatos(Context context) {
        super(context, DATABASE_NAME, null, DATABASE_VERSION);
    }

    @Override
    public void onCreate(SQLiteDatabase db) {
        String CREATE_TABLE_USERS = "CREATE TABLE " + TABLE_USERS + "("
                + COLUMN_ID + " INTEGER PRIMARY KEY,"
                + COLUMN_NAME + " TEXT,"
                + COLUMN_EMAIL + " TEXT,"
                + COLUMN_GENDER + " TEXT,"
                + COLUMN_STATUS + " TEXT" + ")";
        db.execSQL(CREATE_TABLE_USERS);
    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
        db.execSQL("DROP TABLE IF EXISTS " + TABLE_USERS);
        onCreate(db);
    }

    public boolean addUser(Data user) {
        SQLiteDatabase db = this.getWritableDatabase();
        ContentValues values = new ContentValues();
        values.put(COLUMN_ID, user.getId());
        values.put(COLUMN_NAME, user.getName());
        values.put(COLUMN_EMAIL, user.getEmail());
        values.put(COLUMN_GENDER, user.getGender());
        values.put(COLUMN_STATUS, user.getStatus());

        //long result = db.insert(TABLE_USERS, null, values);
        long result = db.insertWithOnConflict(TABLE_USERS, null, values, SQLiteDatabase.CONFLICT_REPLACE);
        db.close();
        return result != -1;
    }

    public Cursor getAllUsers() {
        SQLiteDatabase db = this.getReadableDatabase();
        return db.rawQuery("SELECT * FROM " + TABLE_USERS, null);
    }
}

File: SegundoParcial/app/src/main/java/com/example/segundoparcial/Backend/BaseDatos.kt
package com.example.segundoparcial.Backend

import android.annotation.SuppressLint
import android.content.ContentValues
import android.content.Context
import android.database.Cursor
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper
import kotlin.math.sqrt

data class Puntos1(val id: Int, val x1: Float, val y1: Float, val x2: Float, val y2: Float, val distancia: Float)

class BaseDatos(context: Context): SQLiteOpenHelper(context, BaseDatos.NOMBRE_BASE_DATOS, null, BaseDatos.VERSION_BASE_DATOS) {

    companion object {
        const val NOMBRE_BASE_DATOS = "puntos.db"
        const val VERSION_BASE_DATOS = 1
        const val NOMBRE_TABLA = "puntos"
    }

    override fun onCreate(db: SQLiteDatabase?) {
        val CREAR_TABLA = """
            CREATE TABLE $NOMBRE_TABLA (
                id INTEGER PRIMARY KEY AUTOINCREMENT, 
                x1 FLOAT, 
                y1 FLOAT, 
                x2 FLOAT, 
                y2 FLOAT, 
                distancia FLOAT
            );
        """
        db?.execSQL(CREAR_TABLA)
    }

    override fun onUpgrade(db: SQLiteDatabase?, oldVersion: Int, newVersion: Int) {
        val BORRAR_TABLA = "DROP TABLE IF EXISTS $NOMBRE_TABLA"
        db?.execSQL(BORRAR_TABLA)
        onCreate(db)
    }

    fun addPuntos(puntos: Puntos1): Boolean {
        val db = this.writableDatabase
        val values = ContentValues().apply {
            put("x1", puntos.x1)
            put("y1", puntos.y1)
            put("x2", puntos.x2)
            put("y2", puntos.y2)
            put("distancia", calcularDistancia(puntos.x1, puntos.y1, puntos.x2, puntos.y2))
        }
        val result = db.insert(NOMBRE_TABLA, null, values)
        db.close()
        return result != -1L
    }

    private fun calcularDistancia(x1: Float, y1: Float, x2: Float, y2: Float): Float {
        return sqrt((x2 - x1) * (x2 - x1) + (y2 - y1) * (y2 - y1))
    }

    fun obtenerPuntos(): List<Puntos1> {
        val listaPuntos = mutableListOf<Puntos1>()
        val db = this.readableDatabase
        val query = "SELECT * FROM $NOMBRE_TABLA ORDER BY id DESC"
        val cursor: Cursor = db.rawQuery(query, null)

        if (cursor.moveToFirst()) {
            do {
                val id = cursor.getInt(cursor.getColumnIndexOrThrow("id"))
                val x1 = cursor.getFloat(cursor.getColumnIndexOrThrow("x1"))
                val y1 = cursor.getFloat(cursor.getColumnIndexOrThrow("y1"))
                val x2 = cursor.getFloat(cursor.getColumnIndexOrThrow("x2"))
                val y2 = cursor.getFloat(cursor.getColumnIndexOrThrow("y2"))
                val distancia = cursor.getFloat(cursor.getColumnIndexOrThrow("distancia"))
                val punto = Puntos1(id, x1, y1, x2, y2, distancia)
                listaPuntos.add(punto)
            } while (cursor.moveToNext())
        }

        cursor.close()
        db.close()
        return listaPuntos
    }
}